### PR TITLE
Replace use of depends_on in Traits Properties with observe

### PR DESCRIPTION
**Pull request by rahulporuri**
This PR replaces the use of `depends_on` metadata on Traits Properties with the use of `observe` metadata. Contributes towards #1052

### CODE REVIEW

File: traitsui/ui.py [hunk 1]
```diff
@@ -93,7 +94,12 @@ class UI(HasPrivateTraits):
     history = Any()
 
     #: The KeyBindings object (if any) for this UI:
-    key_bindings = Property(depends_on=["view._key_bindings", "context"])
+    key_bindings = Property(
+        observe=[
+            trait("view._key_bindings", optional=True),
+            "context",
+        ]
+    )
```
Comment (aaronayres35) [inline traitsui/ui.py:102]: I think this use of `trait` may be broken (but break is hidden by `optional=True`)
see https://github.com/enthought/pyface/pull/892 and https://github.com/enthought/pyface/pull/880#discussion_r573969099

File: examples/tutorials/traitsui_4.0/items.py [hunk 2]
```diff
@@ -158,7 +158,7 @@ class LeagueModelView(ModelView):
     got_hit = Button('Got a Hit')
 
     # The total number of hits
-    total_hits = Property(depends_on='model.teams.players.hits')
+    total_hits = Property(observe='model.teams.players.hits')
```
Comment (aaronayres35) [inline examples/tutorials/traitsui_4.0/items.py:161]: I believe we need to insert `.items` on this lists here, eg. `model.teams.items.players.items.hits`

**Review by aaronayres35 — COMMENTED** [review]
Mostly LGTM, but a couple suggested changes.  Most notably in traitsui/ui.py for the use of `trait` on  the extended trait name "view._key_bindings"

I apologize for the non-unified review.  I inadvertently clicked "add single comment", and also "comment" rather than "request changes"

File: traitsui/ui.py (hunk 1 above)
Comment (rahulporuri) [reply traitsui/ui.py]: I removed this to go ahead and merge this PR. We can address the few % of on trait change uses later.

File: examples/tutorials/traitsui_4.0/items.py (hunk 2 above)
Comment (rahulporuri) [reply examples/tutorials/traitsui_4.0/items.py]: same here - i undid this change.

**Review by aaronayres35 — APPROVED** [review]
LGTM

**Pull request merged**